src/FSharp.FGL.IO/GDF.fs

```diff
@@ -0,0 +1,157 @@
+            // Handle quote marks
+            | '\'' when i = chars.Length-1  -> sb.ToString() :: vertexInfos
+            | '\'' when insideQuote         -> stringDeconstruction false  (i+1) vertexInfos (sb)
+            | '\''                          -> stringDeconstruction true (i+1) vertexInfos (sb)
+            // Handle commas
+            | ',' when insideQuote          -> stringDeconstruction insideQuote (i+1) vertexInfos (sb.Append ',')
+            | ','                           -> stringDeconstruction insideQuote (i+1) (sb.ToString() :: vertexInfos) (sb.Clear())
+            // Handle every other symbol
+            | c when i = chars.Length-1     -> sb.Append(c).ToString() :: vertexInfos
+            | c                             -> stringDeconstruction insideQuote (i+1) vertexInfos (sb.Append c)
+
+        stringDeconstruction false 0 [] (StringBuilder())
+        |> List.rev
+        |> List.toArray
+
+    //Returns a appropite function based on the value of the header, to change the corresponding values to their respective TypeDefinitoions type.
+    let private getTypeInfoMapper (headerValue:string) =
+        if      headerValue.Trim().Contains "VARCHAR"   then (fun x -> match x with |" "| "" ->  VARCHAR "DefaultValue"     |_ -> VARCHAR x)
+        elif    headerValue.Trim().Contains "INT"       then (fun x -> match x with |" "| "" ->  INT   1                    |_ -> INT    (int x))
+        elif    headerValue.Trim().Contains "DOUBLE"    then (fun x -> match x with |" "| "" ->  DOUBLE   0.0               |_ -> DOUBLE (float x))
+        elif    headerValue.Trim().Contains "BOOLEAN"   then (fun x -> match x with |" "| "" ->  BOOLEAN true               |"true" -> BOOLEAN      true|"false" -> BOOLEAN      false|_ -> failwith"unknown value in visible")
+        else failwith "unknown typeAnnotation in header"
+
+    //Returns header withput the type annotations or the def>
+    let private headerTrim (header:string) =
+        header.Split ','
+        |>Array.map(fun x ->x.Split ' ')
+        |>Array.map(Array.filter(fun x ->not (x.Contains "VARCHAR")))
+        |>Array.map(Array.filter(fun x ->not (x.Contains "DOUBLE")))
+        |>Array.map(Array.filter(fun x ->not (x.Contains "BOOLEAN")))
+        |>Array.map(Array.filter(fun x ->not (x.Contains "INT")))
+        |>Array.map(Array.item 0)
+        |>Array.map(fun x ->x.Split '>')
+        |>Array.map(Array.filter(fun x ->not (x.Contains "def")))
+        |>Array.map(Array.item 0)
+
+    //Reconstructs the intended Type of var x from TypeInfo         
+    let private reconstructTypeInfo (value:TypeInfo) =    
+           match value with
+               | VARCHAR     x   -> x
+               | BOOLEAN     x   -> sprintf "%b" x
+               | DOUBLE      x   -> sprintf "%f" x
+               | INT         x   -> sprintf "%i" x              
+
+    //Searches for the "name" identifier of the Vertex Map(header,TypeDefiniton), if found returns (string name),Map tupel
```

Comment:
Does the Id necessarily need to be a string? I think you could skip the `reconstructTypeInfo` step here and below

Reply:
Was implemented to make a resulting Graph easier to read, but I admit that it is not necessary.